Sources/SwiftBuildSupport/SwiftBuildSystem.swift

```
@@ -174,6 +174,388 @@ private final class PlanningOperationDelegate: SWBPlanningOperationDelegate, Sen
+            } else {
+                self.observabilityScope.emit(info: "\(info.executionDescription)")
+            }
+
+            if self.logLevel.isVerbose {
+                if let commandLineDisplay = info.commandLineDisplayString {
+                    self.outputStream.send("\(info.executionDescription)\n\(commandLineDisplay)")
+                } else {
+                    self.outputStream.send("\(info.executionDescription)")
+                }
+            }
+            let targetInfo = try buildState.target(for: info)
+            buildSystem.delegate?.buildSystem(buildSystem, willStartCommand: BuildSystemCommand(info, targetInfo: targetInfo))
+            buildSystem.delegate?.buildSystem(buildSystem, didStartCommand: BuildSystemCommand(info, targetInfo: targetInfo))
+        case .taskComplete(let info):
+            let startedInfo = try buildState.completed(task: info)
+
+            // If we've captured the compiler output with formatted diagnostics, emit them.
+//            if let buffer = buildState.dataBuffer(for: startedInfo) {
+            emitDiagnosticCompilerOutput(startedInfo)
+//            }
+
+            if info.result != .success {
+                self.observabilityScope.emit(severity: .error, message: "\(startedInfo.ruleInfo) failed with a nonzero exit code. Command line: \(startedInfo.commandLineDisplayString ?? "<no command line>")")
+            }
+            let targetInfo = try buildState.target(for: startedInfo)
+            buildSystem.delegate?.buildSystem(buildSystem, didFinishCommand: BuildSystemCommand(startedInfo, targetInfo: targetInfo))
+            if let targetName = targetInfo?.targetName {
+                serializedDiagnosticPathsByTargetName[targetName, default: []].append(contentsOf: startedInfo.serializedDiagnosticsPaths.compactMap {
+                    try? Basics.AbsolutePath(validating: $0.pathString)
```

Just let this throw, if this isn't an absolute path something has gone very wrong.